packages/xchain-bitcoin/src/client.ts

```diff
@@ -441,6 +445,62 @@ class Client implements BitcoinClient, XChainClient {
     const txHex = psbt.extractTransaction().toHex() // TX extracted and formatted to hex
     return await blockChair.broadcastTx(this.nodeUrl, txHex, this.nodeApiKey)
   }
+
+  createTxForLedger = async ({
+    asset = AssetBTC,
+    amount,
+    sender,
+    recipient,
+    memo,
+    feeRate,
+  }: LedgerTxParams & { feeRate: FeeRate }): Promise<Utils.LedgerTxInfo> => {
+    await this.scanUTXOs(sender)
+    const balance = await this.getBalance(sender)
+    const btcBalance = balance.find((balance) => balance.asset.symbol === asset.symbol)
+    if (!btcBalance) {
+      throw new Error('No btcBalance found')
+    }
+    if (this.utxos.length === 0) {
+      throw new Error('No utxos to send')
+    }
+    if (!this.validateAddress(recipient)) {
+      throw new Error('Invalid address')
+    }
+    const btcNetwork = this.btcNetwork()
+    const feeRateWhole = Number(feeRate.toFixed(0))
+    const compiledMemo = memo ? Utils.compileMemo(memo) : null
+    const fee = compiledMemo
+      ? Utils.getVaultFee(this.utxos, compiledMemo, feeRateWhole)
+      : Utils.getNormalFee(this.utxos, feeRateWhole)
+    if (amount.amount().plus(fee).isGreaterThan(btcBalance.amount.amount())) {
+      throw new Error('Balance insufficient for transaction')
+    }
+    const psbt = new Bitcoin.Psbt({ network: btcNetwork }) // Network-specific
+    //Inputs
+    this.utxos.forEach((UTXO) =>
+      psbt.addInput({
+        hash: UTXO.hash,
+        index: UTXO.index,
+        witnessUtxo: UTXO.witnessUtxo,
+      }),
+    )
+
+    // Outputs
+    psbt.addOutput({ address: recipient, value: amount.amount().toNumber() }) // Add output {address, value}
+    if (compiledMemo) {
+      // if memo exists
+      psbt.addOutput({ script: compiledMemo, value: 0 }) // Add OP_RETURN {script, value}
+    }
+
+    return {
+      utxos: this.utxos,
+      newTxHex: psbt.data.globalMap.unsignedTx.toBuffer().toString('hex'),
+    }
+  }
+
+  broadcastLedgerTx = async (txHex: string) => {
+    return await blockChair.broadcastTx(this.nodeUrl, txHex, this.nodeApiKey)
+  }
 }
```

Comment:
How about to rename it to `broadcastLedgerTx`, make it `private` and use it in `createTxForLedger` and `transfer`?

Reply:
createTxForLedger is just building tx and it can't be fully built cause ledger sign can't be handled here.
so the aim of this method returns necessary data to where ledger sign can be performed.
Thus, broadcastLedgerTx can't be a part of createTxForLedger

Reply:
Make sense. However, you can still in `transfer`. It's just about to remove duplication of code.